core/sr-primitives/src/lib.rs

```diff
@@ -379,16 +379,17 @@ impl Verify for MultiSignature {
 #[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
 pub struct AnySignature(H512);
 
-/// Public key for any known crypto algorithm.
-#[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Default, Encode, Decode)]
-#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
-pub struct AnySigner(H256);
-
 impl Verify for AnySignature {
-	type Signer = AnySigner;
-	fn verify<L: Lazy<[u8]>>(&self, mut msg: L, signer: &AnySigner) -> bool {
-		runtime_io::sr25519_verify(self.0.as_fixed_bytes(), msg.get(), &signer.0.as_bytes()) ||
-			runtime_io::ed25519_verify(self.0.as_fixed_bytes(), msg.get(), &signer.0.as_bytes())
+	type Signer = sr25519::Public;
```

Comment:
I've used `AnySigner` in #2033, I think it's a bit more explicit, can later work on merging this.